data-prepper-plugins/kafka-plugins/src/main/java/org/opensearch/dataprepper/plugins/kafka/consumer/KafkaSourceCustomConsumer.java

```diff
@@ -357,8 +366,8 @@ private <T> void iterateRecordPartitions(ConsumerRecords<String, T> records, fin
                                              Map<TopicPartition, CommitOffsetRange> offsets) throws Exception {
         for (TopicPartition topicPartition : records.partitions()) {
             final long partitionEpoch = getPartitionEpoch(topicPartition);
-            if (partitionEpoch == 0) {
-                LOG.info("Skipping partition {}, lost ownership", topicPartition);
+            if (acknowledgementsEnabled && partitionEpoch == 0) {
+                //ToDo: Add metric
```

Comment:
Would the metric here be useful for debugging? I'm wondering if it's worth just adding now

Reply:
I didn't have time to add a new metric. I'll see if I can add now, but there are other metrics and logs from which it can be inferred if we are hitting this condition. Ideally, it should not happen and an assert is more appropriate here.